powerdns/models/powerdns.py

```diff
@@ -278,7 +278,7 @@ def owners(self):
     @property
     def service_owners(self):
         """Returns owners by service unless direct_owners are set"""
-        owners = self.direct_owners.through.objects
+        owners = self.direct_owners.through.objects.filter(domain=self)
```

Comment:
After simialr case last time, i was trying to did this but it didn't work, i think.
Isn't this accessible when `related_name` is added?